CMakeLists.txt

```diff
@@ -16,26 +26,21 @@ option(USE_BOOST_REGEX "Replace std::regex with Boost.Regex" OFF)
 # C++ standard
 set(CMAKE_CXX_STANDARD_REQUIRED ON)
 set(CMAKE_CXX_EXTENSIONS OFF)
-if(NOT CMAKE_CXX_STANDARD OR CMAKE_CXX_STANDARD LESS 11)
-	set(CMAKE_CXX_STANDARD 11)
+if(NOT CMAKE_CXX_STANDARD OR CMAKE_CXX_STANDARD LESS 17)
+    set(CMAKE_CXX_STANDARD 17)
```

Comment:
1) if you're modernizing, this needs some rewriting. This isn't modern cmake as it's global instead of target specific, no?
2) While switching to c++17 might be useful for some users, mixing it with this pull request makes it hard to accept the rest of the benefits of this change, as dropping all support for older compilers is probably not a desirable change.

Reply:
The current standard is c++20
c++14 is deprecated
and c++11 is buggy

Reply:
I just saw this .. this library is intended to work with C++11 and has no C++17 features.

What do you mean by "C++11" is buggy?

Reply:
Maybe we should just remove the default completely and let it be "whatever the compiler wants to be default", and then we can have our CI system explicitly set it to other values (including 11 and 23)?